### ISSUE TITLE
PLM NAKs not respected

### ISSUE THREAD

**Issue by rcombs** [automategreen/home-controller#87]
PLMs can sometimes return 0x15 bytes (NAK) when a command is sent too quickly after a previous command. Currently, this is ignored, resulting in the default 5-second timeout passing before the command is resent. Instead, we can catch the NAKs and respond by temporarily resetting the retry timeout to something short (I used 10ms with 2x exponential backoff) without consuming one of the global retries.

I added this code to the `Insteon` constructor:
`this.nakTimeout = 10;`
Adjusted this line in `sendCommand`:
`var status = { command: cmd, deferred: deferred, timeout: timeout, nakTimeout: this.nakTimeout };`
and added this near the top of `_checkStatus` (before the length <= 4 check):
```
if (raw.substr(0, 2) == '15' && this.status && this.timeout) {
  // Got a PLM NAK; retry our command, if applicable
  var gw = this;
  var nakTimeout = this.status.nakTimeout;
  this.status.nakTimeout *= 2; // Exponential backoff
  clearTimeout(this.timeout);
  this.timeout = setTimeout(function () {
    sendCommandTimout(gw, gw.status.timeout, gw.commandRetries);
  }, nakTimeout);
  this.buffer = raw.substr(2);
  return MESSAGE_PROCESSED;
}
```

This dramatically improves performance in home-controller for me (on a Hub Pro with a fair number of devices).

**Comment by siburny** [comment]
Thank you for your contribution. I think this patch will help alleviate the problem mentioned in https://github.com/automategreen/home-controller/issues/83.

Would you like to submit a PR? or I can just do a commit your self.

**Comment by rcombs** [comment]
I think this should also help with #39.

My setup isn't really in a convenient position to make a PR; if you could commit that'd be great.

**Comment by rcombs** [comment]
Ping on this.

**Comment by siburny** [comment]
Fixed in e1e329191e845b7b42529ead06e9f385a752bd5c
